packages/base-types/src/models/project/index.ts

```diff
@@ -32,6 +32,44 @@ export interface AIAssistSettings {
   generativeTasks?: boolean;
 }
 
+export enum KnowledgeBaseDocumentType {
+  PDF = 'pdf',
+  TEXT = 'text',
+  URL = 'url',
+}
+
+export interface KnowledgeBasePDF {
+  type: KnowledgeBaseDocumentType.PDF;
+  name: string;
+}
+
+export interface KnowledgeBaseText {
+  type: KnowledgeBaseDocumentType.TEXT;
+  name: string;
+}
+
+export interface KnowledgeBaseURL {
+  type: KnowledgeBaseDocumentType.URL;
+  name: string;
+  url: string;
+}
+
+export enum KnowledgeBaseDocumentStatus {
+  UPLOADING = 'UPLOADING',
+  PARSING = 'PARSING',
+  EMBEDDING = 'EMBEDDING',
+  FINISHED = 'FINISHED',
+}
+
+export interface KnowledgeBaseDocument {
+  data: KnowledgeBasePDF | KnowledgeBaseText | KnowledgeBaseURL;
```

Comment:
These subtypes should ideally inherit from a common interface, like `KnowledgeBaseData`. 

```ts
export interface KnowledgeBaseData {
  type: KnowledgeBaseDocumentType;
  name: string 
}

export interface KnowledgeBasePDF extends KnowledgeBaseData {
  type: KnowledgeBaseDocumentType.PDF;
}

export interface KnowledgeBaseText extends KnowledgeBaseData {
  type: KnowledgeBaseDocumentType.TEXT;
}

export interface KnowledgeBaseURL extends KnowledgeBaseData {
  type: KnowledgeBaseDocumentType.URL;
  url: string;
}
```